### PR TITLE
feat: New issue templates (following new labelling process)

### PR DESCRIPTION
**Pull request by loujaybee**
Updates following internal RFC on labelling and process updates. 

**Five new templates:** 
1. **Content suggestion** - issues for product docs
1. **Content bug** - bug on existing page with content
1. **Tech suggestion** - for website team
1. **Tech bug** - bug on existing page with css/tech
1. **Unknown suggestion** - user is unsure

**Template labels:** 
1. All issues with templates have `type` and `section` labels
1. `team`, `priority`, `status` to add during triaging

Based on new labelling proposal ([internal RFC](https://www.notion.so/gitpod/Content-Backlogs-What-Content-Suggestions-Go-Where-8eb6ed9e97824422bdfaca9343af4e90)): 

image

<a href="https://gitpod-staging.com/#https://github.com/gitpod-io/website/pull/2902"></a>

<a href="https://gitpod.io/#https://github.com/gitpod-io/website/pull/2902"></a>

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/content-suggestion.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+name: Content Suggestion
```
Comment (loujaybee) [inline .github/ISSUE_TEMPLATE/content-suggestion.yml:1]: Do file names have to be caps? I'm not sure?

File: .github/ISSUE_TEMPLATE/tech-suggestion.yml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+name: Technical Suggestion
+description: Request a technical change to the website, e.g. new component, update to navigation, etc.
+labels: ["section: docs-non-content"]
```
Comment (loujaybee) [inline .github/ISSUE_TEMPLATE/tech-suggestion.yml:3]: Look good to you @Cahllagerfeld ?

File: .github/ISSUE_TEMPLATE/content-suggestion.yml [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+description: An update to documentation content (e.g. not a tech change like a component)
+labels: ["section: docs-content"]
+body:
+  - type: textarea
+    attributes:
+      label: Description
+    validations:
+      required: true
```
Comment (loujaybee) [inline .github/ISSUE_TEMPLATE/content-suggestion.yml:9]: Chose to simplify here, should only be internal people, the strict format was adding friction for little value IMO. 

A freeform text that allows any input should be fine, we don't get masses of suggestions anyway.

File: .github/ISSUE_TEMPLATE/content-suggestion.yml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+name: Content Suggestion
+description: An update to documentation content (e.g. not a tech change like a component)
+labels: ["section: docs-content"]
```
Comment (loujaybee) [inline .github/ISSUE_TEMPLATE/content-suggestion.yml:3]:
```suggestion
labels: ["section: docs-content", "type: idea"]
```

File: README.md [hunk 5]
```diff
@@ -7,33 +7,31 @@
...
-
-[![Open in Gitpod](https://www.gitpod.io/svg/open-in-gitpod.svg)](https://gitpod.io/#https://github.com/gitpod-io/website)
+## Issues
 
+**Mandatory labels:**
 
-## CLI Commands
+- A `team: x` - Which team the issue belongs to
+- A `section: x` - Which area of docs it corresponds to (helps understand ownership)
+- A `type: x` - e.g. `bug`, `idea`, `question`, `backlog`
```
Comment (loujaybee) [inline README.md:16]:
```suggestion
- A `type: x` - e.g. `bug`, `suggestion`, `question`
```

File: .github/ISSUE_TEMPLATE/tech-suggestion.yml (hunk 2 above)
Comment (loujaybee) [inline .github/ISSUE_TEMPLATE/tech-suggestion.yml:3]:
```suggestion
labels: ["section: docs-non-content", "type: suggestion"]
```

File: .github/ISSUE_TEMPLATE/content-suggestion.yml (hunk 1 above)
Comment (Siddhant-K-code) [reply .github/ISSUE_TEMPLATE/content-suggestion.yml]: Yeah, Filenames should be in all caps. (related [docs](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#:~:text=create%20a%20file%20called%20.github/ISSUE_TEMPLATE/FORM%2DNAME.yml%2C%20replacing%20FORM%2DNAME))

resolved in [**`2efaca4`**](https://github.com/gitpod-io/website/pull/2902/commits/2efaca41a8d3f22db2b370bc4226ac7b50fbaedb)

File: .github/ISSUE_TEMPLATE/TECHNICAL-BUG.yml [hunk 6]
```diff
@@ -0,0 +1,17 @@
+name: 🐞 Technical Bug
+description: A technical bug, such as wrong font, navigation issue, related to an existing page.
+labels: ["type: bug", "section: docs-non-content"]
```
Comment (Siddhant-K-code) [inline .github/ISSUE_TEMPLATE/TECHNICAL-BUG.yml:3]: I think we can add another label called [`needs-technical-help`](https://github.com/gitpod-io/website/issues?q=is%3Aopen+is%3Aissue+label%3Aneeds-technical-help) to be more specific. @Cahllagerfeld WDYT?

File: README.md [hunk 7]
```diff
@@ -7,33 +7,31 @@
...
-### Generate JetBrains Docs
-
-```sh
-npm run generate:jb
-```
+## CLI Commands
 
-## Technology stack
+- `npm run generate:blog-post` - Generate blog post
+- `npm run generate:jb` - Generate JetBrains pages
```
Comment (Siddhant-K-code) [inline README.md:23]: Can we revert the changes for this part? GitHub provides a direct copy-paste button if you write it with ```.

For Example:

```text
HELLO
```

`HELLO`

File: README.md [hunk 8]
```diff
@@ -7,33 +7,31 @@
...
 - [SvelteKit](https://kit.svelte.dev)
 - [mdsvex](https://mdsvex.com)
 - [Tailwind CSS](https://tailwindcss.com)
-    > **Note**: Custom CSS was used initially. This is considered legacy and any PR going forward requires Tailwind CSS.
-
 - [Netlify](https://www.netlify.com)
+
+## Contribute
+
+Contributing is as easy as starting a new workspace:
```
Comment (Cahllagerfeld) [inline README.md:35]: Do we need to mention the CLA-process here?

File: .github/ISSUE_TEMPLATE/TECHNICAL-SUGGESTION.yml [hunk 9]
```diff
@@ -0,0 +1,9 @@
+name: ✨ Technical Suggestion
+description: Request a technical change to the website, e.g. new component, update to navigation, etc.
+labels: ["section: docs-non-content", "type: suggestion"]
```
Comment (Cahllagerfeld) [inline .github/ISSUE_TEMPLATE/TECHNICAL-SUGGESTION.yml:3]: I'm not sure yet if I'm a fan of solely using the `docs-non-content`-label. I feel we're describing a thing, by saying what it is not. Maybe we should be a bit more specific and add another `tech`-label or something like Siddhant suggested 🤔 what do you think @loujaybee

File: .github/ISSUE_TEMPLATE/CONTENT-BUG.yml [hunk 10]
```diff
@@ -0,0 +1,17 @@
+name: 🐛 Content Bug
+description: A content related issue such as a typo or error in existing content
+labels: ["type: bug", "section: docs-content"]
```
Comment (chrifro) [inline .github/ISSUE_TEMPLATE/CONTENT-BUG.yml:3]: Does this only relate to /docs or generally on the website? It's not clear to me right now. 
I'd suggest keeping it general and removing the docs label. WDYT?

File: .github/ISSUE_TEMPLATE/CONTENT-SUGGESTION.yml [hunk 11]
```diff
@@ -0,0 +1,9 @@
+name: 📝 Content Suggestion
+description: An update to documentation content (e.g. not a tech change like a component)
```
Comment (chrifro) [inline .github/ISSUE_TEMPLATE/CONTENT-SUGGESTION.yml:2]: If this only relates to docs I'd add that in the title e.g. "Docs content suggestion"

File: .github/ISSUE_TEMPLATE/TECHNICAL-BUG.yml (hunk 6 above)
Comment (chrifro) [reply .github/ISSUE_TEMPLATE/TECHNICAL-BUG.yml:3]: I'd say the `type: needs-technical-help` label should not be added automatically but used by us once an issue is prioritized but we are stuck and need to acquire more resources to resolve it.

**Review by chrifro — APPROVED** [review]

File: README.md (hunk 7 above)
Comment (loujaybee) [reply README.md:23]: Yeah, can you? 😁 

Could we use the smallest possible header?

File: README.md (hunk 8 above)
Comment (loujaybee) [reply README.md:35]: This is not new content, can we consider out of scope? 🙏

File: .github/ISSUE_TEMPLATE/TECHNICAL-SUGGESTION.yml (hunk 9 above)
Comment (loujaybee) [reply .github/ISSUE_TEMPLATE/TECHNICAL-SUGGESTION.yml:3]: I don't currently have a better suggestion, given that this PR implements the current process as is currently, I'd suggest we merge, and then can discuss and follow up with improvements? 🙏

**Comment by loujaybee** [comment]
Will swap assignee for @Siddhant-K-code to close this off (small final tweaks!)

Thanks @chrifro / @Cahllagerfeld for the reviews 🙏

**Comment by loujaybee** [comment]
CC: @jldec, @andré, @lucasvaltl, @mbrevoort - for visibility, wanted to share these changes / updates to how issues are raised in this repo, and the GitHub label changes.

**Comment by loujaybee** [comment]
✅  - if this weren't my own PR, I would approve ! :shipit:

**Review by Siddhant-K-code — APPROVED** [review]
LGTM ✨ 

If there would be any feedback, we can iterate on it later on. We can :shipit: right away!
